runtime/compiler/env/ProcessorDetection.cpp

```diff
@@ -653,62 +653,12 @@ TR_J9VM::initializeProcessorType()
    if (TR::Compiler->target.cpu.isZ())
       {
 #if defined(TR_HOST_S390)
-      initializeS390ProcessorFeatures();
-
 #if defined(J9ZOS390)
       // Cache whether current process is running in Supervisor State (i.e. Control Region of WAS).
```

Comment:
So where are things initialized now? Is it in OMR?

Reply:
Step 1. OMR detects and initializes the CPU using omr port library.

```cpp
void
OMR::CompilerEnv::initializeHostEnvironment()
   {

   // Host processor bitness
   //
#ifdef TR_HOST_64BIT
   host.setBitness(TR::bits_64);
#elif TR_HOST_32BIT
   host.setBitness(TR::bits_32);
#else
   host.setBitness(TR::bits_unknown);
#endif

   // Initialize the host CPU by querying the host processor
   //
   host.cpu = TR::CPU::detect(TR::Compiler->omrPortLib);
```

Step 2. OpenJ9 calls `J9::Z::CPU::applyUserOptions()` to apply command line options from user and initialize the J9-specific features that couldn't be done in OMR.

```cpp
void
TR_J9VM::initializeProcessorType()
   {
   TR_ASSERT(_compInfo,"compInfo not defined");
   TR::Compiler->target.cpu.applyUserOptions();
```

Reply:
`applyUserOptions()` mirrors the things that were done in `initializeS390ProcessorFeatures()`

Reply:
Thanks for the explanation! We'll wait until the OMR change has propagated.